napari/_qt/qt_main_window.py

```diff
@@ -265,6 +265,10 @@ def show(self):
         self._qt_window.show()
         # make sure window is not hidden, e.g. by browser window in Jupyter
         self._qt_window.raise_()
+        # Update to MAX_TEXTURE_SIZE needs to come after window is shown
+        self.qt_viewer.MAX_TEXTURE_SIZE = self.qt_viewer.canvas.context.shared.parser.capabilities[
+            'max_texture_size'
+        ]
```

Comment:
@sofroniewn why does this need to come after the window? It seems that vispy has some ways to introspect this value, see:

https://github.com/vispy/vispy/issues/1141

(And note from that thread that this value is different for 2D and 3D...)

Reply:
`sys_info` actually creates a little fake canvas in order to do this - https://github.com/vispy/vispy/blob/a6ec64bb7b56739b463e3c16245a31940ca4f5f7/vispy/util/config.py#L431 which we could do too once on start-up - though probably don't want to put inside our "model" code. I have an idea on how to at least improve what we have now significantly.

Thanks for catching `GL_MAX_3D_TEXTURE_SIZE` I can work with that too

Reply:
so vispy does not actually expose `GL_MAX_3D_TEXTURE_SIZE` - I'm not quite sure why, I need to do some investigating before asking

Reply:
fixed, though see note about `GL_MAX_3D_TEXTURE_SIZE` not being exposed